shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp

```diff
@@ -32,8 +32,9 @@ namespace shared_model {
       template <typename CommandType>
       explicit AddAssetQuantity(CommandType &&command)
           : command_(std::forward<CommandType>(command)),
-            add_asset_quantity_(
-                [this] { return command_->add_asset_quantity(); }),
+            add_asset_quantity_([this]() -> decltype(auto) {
```

Comment:
This lambda should return reference, since lazy template type is reference to const. This is not redundant, since it allows to deduce reference. https://isocpp.org/wiki/faq/cpp14-language#decltype-auto

Reply:
This looks not so obvious, maybe provide meeting about that?